tests/system/action/poll/test_vote.py

```diff
@@ -577,3 +596,1207 @@ def test_default_vote_weight(self) -> None:
+        option1 = self.get_model("option/1")
+        option2 = self.get_model("option/2")
+        option3 = self.get_model("option/3")
+        self.assertEqual(option1.get("yes"), "4.200000")
+        self.assertEqual(option1.get("no"), "0.000000")
+        self.assertEqual(option1.get("abstain"), "0.000000")
+        self.assertEqual(option2.get("yes"), "0.000000")
+        self.assertEqual(option2.get("no"), "4.200000")
+        self.assertEqual(option2.get("abstain"), "0.000000")
+        self.assertEqual(option3.get("yes"), "0.000000")
+        self.assertEqual(option3.get("no"), "0.000000")
+        self.assertEqual(option3.get("abstain"), "4.200000")
+
+    def test_change_vote(self) -> None:
+        self.start_poll()
+        response = self.request(
+            "poll.vote",
+            {"value": {"1": "Y"}, "id": 1, "user_id": 1},
+        )
+        response = self.request(
+            "poll.vote",
+            {"value": {"1": "N"}, "id": 1, "user_id": 1},
+        )
+        self.assert_status_code(response, 400)
+        self.assert_model_not_exists("vote/2")
+        vote = self.get_model("vote/1")
+        self.assertEqual(vote.get("value"), "Y")
+
+    def test_too_many_options(self) -> None:
+        self.start_poll()
+        response = self.request(
+            "poll.vote",
+            {"value": {"1": "Y", "3": "N"}, "id": 1, "user_id": 1},
+        )
+        self.assert_status_code(response, 400)
+        self.assert_model_not_exists("vote/1")
+
+    def test_wrong_options(self) -> None:
+        self.start_poll()
+        response = self.request(
+            "poll.vote",
+            {"value": {"1": "Y", "3": "N"}, "id": 1, "user_id": 1},
+        )
+        self.assert_status_code(response, 400)
+        self.assert_model_not_exists("vote/1")
+
+    @pytest.mark.skip()
+    def test_no_permissions(self) -> None:
+        self.start_poll()
+        # self.make_admin_delegate()
+        response = self.request(
+            "poll.vote",
+            {"value": {"1": "Y"}, "id": 1, "user_id": 1},
+        )
+        self.assert_status_code(response, 400)
+        self.assert_model_not_exists("vote/1")
+
+    def test_anonymous(self) -> None:
+        self.start_poll()
+        response = self.request(
+            "poll.vote",
+            {"value": {"1": "Y"}, "id": 1, "user_id": 0},
+        )
+        self.assert_status_code(response, 400)
+        self.assert_model_not_exists("vote/1")
+
+    def test_vote_not_present(self) -> None:
+        self.start_poll()
+        self.update_model("user/1", dict(is_present_in_meeting_ids=[]))
+        response = self.request(
+            "poll.vote",
+            {"value": {"1": "Y"}, "id": 1, "user_id": 1},
+        )
+        self.assert_status_code(response, 400)
+        self.assert_model_not_exists("vote/1")
+
+    def test_wrong_state(self) -> None:
+        response = self.request("poll.vote", {"value": {}, "id": 1, "user_id": 1})
```

Comment:
This does not have something to do with wrong state, does it?

Reply:
It is missing the start_poll, so it fails because of the Created state.

Reply:
I have improved the test, so it is more obvious.